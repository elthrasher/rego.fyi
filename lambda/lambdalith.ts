import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';

export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
  const { httpMethod, path } = event;
  return {
    body: JSON.stringify({ httpMethod, path }),
    headers: { 'Access-Control-Allow-Credentials': true, 'Access-Control-Allow-Origin': 'https://rego.fyi' },
    statusCode: 200,
  };
};
