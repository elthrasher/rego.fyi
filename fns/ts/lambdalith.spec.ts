import { APIGatewayProxyEvent } from 'aws-lambda';

import { handler } from './lambdalith';

describe('lambdalith tests', () => {
  test('should return a response', async () => {
    const response = await handler({ httpMethod: 'GET', path: '/orders' } as APIGatewayProxyEvent);
    expect(response).toMatchSnapshot();
  });
});
