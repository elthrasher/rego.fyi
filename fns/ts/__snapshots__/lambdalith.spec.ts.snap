// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lambdalith tests should return a response 1`] = `
Object {
  "body": "{\\"httpMethod\\":\\"GET\\",\\"path\\":\\"/orders\\"}",
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Origin": "https://rego.fyi",
  },
  "statusCode": 200,
}
`;
